logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/message/MessageSenderTest.kt

```diff
@@ -0,0 +1,332 @@
+
+    @Mock
+    private val mlsMessageCreator: MLSMessageCreator = mock(MLSMessageCreator::class)
+
+    @Mock
+    private val syncManager = configure(mock(SyncManager::class)) { stubsUnitByDefault = true }
+
+    private lateinit var messageSender: MessageSender
+
+    @BeforeTest
+    fun setup() {
+        messageSender = MessageSenderImpl(
+            messageRepository = messageRepository,
+            conversationRepository = conversationRepository,
+            syncManager = syncManager,
+            messageSendFailureHandler = messageSendFailureHandler,
+            sessionEstablisher = sessionEstablisher,
+            messageEnvelopeCreator = messageEnvelopeCreator,
+            mlsMessageCreator = mlsMessageCreator
+        )
+    }
+
+    @Test
+    fun givenAllStepsSucceed_WhenSendingOutgoingMessage_ThenReturnSuccess() {
+        runTest {
+            //given
+            setupGivenSuccessResults(
+                getMessageById = true,
+                getConversationProtocol = true,
+                getConversationsRecipient = true,
+                prepareRecipientsForNewOutGoingMessage = true,
+                createOutgoingEnvelope = true,
+                sendEnvelope = true,
+                updateMessageStatus = true
+            )
+            //when
+            val result = messageSender.trySendingOutgoingMessageById(ConversationId("testId", "testDomain"), "testId")
+            //then
+            assertIs<Either.Right<Unit>>(result)
+        }
+    }
+
+    @Test
+    fun givenGettingConversationProtocolFails_WhenSendingOutgoingMessage_ThenReturnFailureAndSetMessageStatusToFailed() {
+        runTest {
+            //given
+            setupGivenSuccessResults(
+                getMessageById = true,
+                getConversationProtocol = false,
+                getConversationsRecipient = true,
+                prepareRecipientsForNewOutGoingMessage = true,
+                createOutgoingEnvelope = true,
+                sendEnvelope = true,
+                updateMessageStatus = true
+            )
+            //when
+            val result = messageSender.trySendingOutgoingMessageById(ConversationId("testId", "testDomain"), "testId")
+            //then
+            verify(messageRepository)
+                .suspendFunction(messageRepository::updateMessageStatus)
+                .with(eq(MessageEntity.Status.FAILED), anything(), anything())
+                .wasInvoked(Times(1))
+
+            assertIs<Either.Left<Unit>>(result)
+        }
+    }
+
+    @Test
+    fun givenGettingConversationRecipientsFails_WhenSendingOutgoingMessage_ThenReturnFailureAndSetMessageStatusToFailed() {
+        runTest {
+            //given
+            //given
+            setupGivenSuccessResults(
+                getMessageById = true,
+                getConversationProtocol = false,
+                getConversationsRecipient = true,
```

Comment:
```suggestion
                getConversationProtocol = true,
                getConversationsRecipient = false,
```

I guess it was supposed to be like this?